owncloudApp/src/main/java/com/owncloud/android/ui/activity/PassCodeActivity.kt

```diff
@@ -0,0 +1,425 @@
+        if (ACTION_CHECK == intent.action) {
+            /// this is a pass code request; the user has to input the right value
+            mPassCodeHdr.setText(R.string.pass_code_enter_pass_code)
+            mPassCodeHdrExplanation.setVisibility(View.INVISIBLE)
+            setCancelButtonEnabled(false) // no option to cancel
+        } else if (ACTION_REQUEST_WITH_RESULT == intent.action) {
+            if (savedInstanceState != null) {
+                mConfirmingPassCode = savedInstanceState.getBoolean(KEY_CONFIRMING_PASSCODE)
+                mPassCodeDigits = savedInstanceState.getStringArray(KEY_PASSCODE_DIGITS)!!
+            }
+            if (mConfirmingPassCode) {
+                //the app was in the passcodeconfirmation
+                requestPassCodeConfirmation()
+            } else {
+                /// pass code preference has just been activated in Preferences;
+                // will receive and confirm pass code value
+                mPassCodeHdr.setText(R.string.pass_code_configure_your_pass_code)
+                //mPassCodeHdr.setText(R.string.pass_code_enter_pass_code);
+                // TODO choose a header, check iOS
+                mPassCodeHdrExplanation.setVisibility(View.VISIBLE)
+                setCancelButtonEnabled(true)
+            }
+        } else if (ACTION_CHECK_WITH_RESULT == intent.action) {
+            /// pass code preference has just been disabled in Preferences;
+            // will confirm user knows pass code, then remove it
+            mPassCodeHdr.setText(R.string.pass_code_remove_your_pass_code)
+            mPassCodeHdrExplanation.setVisibility(View.INVISIBLE)
+            setCancelButtonEnabled(true)
+        } else {
+            throw IllegalArgumentException(R.string.illegal_argument_exception_message.toString() + " ")
+        }
+        setTextListeners()
+    }
+
+    private fun inflatePasscodeTxtLine(passcodeLockLayout: LinearLayout) {
+        val passcodeTxtLayout = findViewById<LinearLayout>(R.id.passCodeTxtLayout)
+        for (i in 0 until numberOfPassInputs) {
+            val txt = layoutInflater.inflate(R.layout.passcode_edit_text, passcodeTxtLayout, false) as EditText
+            passcodeTxtLayout.addView(txt)
+            mPassCodeEditTexts[i] = txt
+        }
+        mPassCodeEditTexts[0]!!.requestFocus()
+        window.setSoftInputMode(
+            WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE
+        )
+    }
+
+    /**
+     * Enables or disables the cancel button to allow the user interrupt the ACTION
+     * requested to the activity.
+     *
+     * @param enabled       'True' makes the cancel button available, 'false' hides it.
+     */
+    protected fun setCancelButtonEnabled(enabled: Boolean) {
+        if (enabled) {
+            mBCancel.visibility = View.VISIBLE
+            mBCancel.setOnClickListener { finish() }
+        } else {
+            mBCancel.visibility = View.GONE
+            mBCancel.visibility = View.INVISIBLE
```

Comment:
Is there any reason to change visibility first to `GONE` and then to `INVISIBLE`?